internal/controller/dnsrecord_controller.go

```diff
@@ -91,19 +86,19 @@ func (r *DNSRecordReconciler) Reconcile(ctx context.Context, req ctrl.Request) (
 	dnsRecord := previous.DeepCopy()
 
 	if dnsRecord.DeletionTimestamp != nil && !dnsRecord.DeletionTimestamp.IsZero() {
-		if err := r.ReconcileHealthChecks(ctx, dnsRecord); client.IgnoreNotFound(err) != nil {
+		if err = r.ReconcileHealthChecks(ctx, dnsRecord); client.IgnoreNotFound(err) != nil {
 			return ctrl.Result{}, err
 		}
-		requeueTime, err := r.deleteRecord(ctx, dnsRecord)
+		hadChanges, err := r.deleteRecord(ctx, dnsRecord)
```

Comment:
```suggestion
		appliedChanges, err := r.deleteRecord(ctx, dnsRecord)
```

Looks to me like it tells you if changes were applied (calculated changes and successfully applied them) rather than just if there were any. Maybe the name of this variable should reflect that?

Reply:
If I'm not mistaken it tells if there were any changes in the plan regardless of how successful it was in applying them